src/components/big-interactive-pages/editor.tsx

```diff
@@ -39,34 +39,49 @@ const heightMargin = 130
 const foldAllTemplateLiterals = () => {
 	if (!codeMirror.value) return
 	const code = codeMirror.value.state.doc.toString() ?? ''
-	const matches = [ ...code.matchAll(/(map|bitmap|tune)`[\s\S]*?`/g) ];
-	collapseRanges(codeMirror.value, matches.map((match) => [ match.index!, match.index! + 1 ]))
+	const matches = [...code.matchAll(/(map|bitmap|tune)`[\s\S]*?`/g)];
+	collapseRanges(codeMirror.value, matches.map((match) => [match.index!, match.index! + 1]))
 }
 
 let lastSavePromise = Promise.resolve()
 let saveQueueSize = 0
 export const saveGame = debounce(800, (persistenceState: Signal<PersistenceState>, code: string) => {
 	const doSave = async () => {
 		let isError = false
-		try {
-			const game = (persistenceState.value.kind === 'PERSISTED' && persistenceState.value.game !== 'LOADING') ? persistenceState.value.game : null
-			const res = await fetch('/api/games/save', {
-				method: 'POST',
-				headers: { 'Content-Type': 'application/json' },
-				body: JSON.stringify({ code, gameId: game?.id, tutorialName: game?.tutorialName, tutorialIndex: game?.tutorialIndex })
-			})
-			if (!res.ok) throw new Error(`Error saving game: ${await res.text()}`)
-		} catch (error) {
-			console.error(error)
-			isError = true
+		const attemptSaveGame = async () => {
+			try {
+				const game = (persistenceState.value.kind === 'PERSISTED' && persistenceState.value.game !== 'LOADING') ? persistenceState.value.game : null
+				const res = await fetch('/api/games/save', {
+					method: 'POST',
+					headers: { 'Content-Type': 'application/json' },
+					body: JSON.stringify({ code, gameId: game?.id, tutorialName: game?.tutorialName, tutorialIndex: game?.tutorialIndex })
+				})
+				if (!res.ok) throw new Error(`Error saving game: ${await res.text()}`)
+				isError = false
+			} catch (error) {
+				console.error(error)
+				isError = true
+
+				persistenceState.value = {
+					...persistenceState.value,
+					cloudSaveState: 'ERROR'
+				} as any;
+			}
+		}
+
+		await attemptSaveGame();
+		while (isError) {
```

Comment:
The usage of this 'isError' variable is slightly evil.  Couldn't the error state just return from the attemptSaveGame method instead of managing mutable state, and wouldn't that be cleaner?

i.e. while(!await attemptSaveGame()) {